contracts/BaseWrapper.sol

```diff
@@ -0,0 +1,197 @@
+// SPDX-License-Identifier: GPL-3.0
+pragma solidity ^0.6.12;
+pragma experimental ABIEncoderV2;
+
+import {ERC20} from "@openzeppelin/contracts/token/ERC20/ERC20.sol";
+import {SafeMath} from "@openzeppelin/contracts/math/SafeMath.sol";
+
+import {RegistryAPI, VaultAPI} from "./BaseStrategy.sol";
+
+abstract contract BaseWrapper {
+    using SafeMath for uint256;
+
+    ERC20 public token;
+
+    // Reduce number of external calls (SLOADs stay the same)
+    VaultAPI[] private _cachedVaults;
+
+    RegistryAPI public registry;
+
+    constructor(address _token, address _registry) public {
+        token = ERC20(_token);
+        // v2.registry.ychad.eth
+        registry = RegistryAPI(_registry);
+    }
+
+    function setRegistry(address _registry) external {
+        require(msg.sender == registry.governance());
+        // In case you want to override the registry instead of re-deploying
+        registry = RegistryAPI(_registry);
+    }
+
+    function bestVault() public virtual view returns (VaultAPI) {
+        return VaultAPI(registry.latestVault(address(token)));
+    }
+
+    function allVaults() public virtual view returns (VaultAPI[] memory) {
+        uint256 cache_length = _cachedVaults.length;
+        uint256 num_deployments = registry.nextDeployment(address(token));
+
+        // Use cached
+        if (cache_length == num_deployments) {
+            return _cachedVaults;
+        }
+
+        VaultAPI[] memory vaults = new VaultAPI[](num_deployments);
+
+        for (uint256 deployment_id = 0; deployment_id < cache_length; deployment_id++) {
+            vaults[deployment_id] = _cachedVaults[deployment_id];
+        }
+
+        for (uint256 deployment_id = cache_length; deployment_id < num_deployments; deployment_id++) {
+            vaults[deployment_id] = VaultAPI(registry.vaults(address(token), deployment_id));
+        }
+
+        return vaults;
+    }
+
+    function _updateVaultCache(VaultAPI[] memory vaults) internal {
+        if (vaults.length > _cachedVaults.length) {
+            _cachedVaults = vaults;
+        }
+    }
+
+    function totalVaultBalance(address account) public view returns (uint256 balance) {
+        VaultAPI[] memory vaults = allVaults();
+
+        for (uint256 id = 0; id < vaults.length; id++) {
+            balance = balance.add(vaults[id].balanceOf(account).mul(vaults[id].pricePerShare()).div(10**uint256(vaults[id].decimals())));
```

Comment:
I was brain storming the feasibility of the following attack:
-flash loan
-deposit portion in affiliate wrapper at pricePerShare 1 in underlying vault and get 100 shares
-deposit into Vaults directly with another portion and increase pricePerShare 1.2
-withdraw from wrapper 100 shares now valued at 1.2, do i get more tokens?
-withdraw from vault back funds

Im half thinking it doesnt work, is it feasible or worth testing/validating?

Reply:
talked with doggie, and step 3 is not feasible, since he correctly points out that cannot bump the price per share without a harvest just by deposits